python/pyspark/sql/tests.py

```diff
@@ -30,6 +30,19 @@
 import functools
 import time
 import datetime
+import traceback
+
+if sys.version_info[:2] <= (2, 6):
```

Comment:
Ah, hmm.. it should specifically check Python 2.6 <= as unittest2 is unittest from Python 2.7. To check minor versions, I think we should "extract" or compare with raw string `'2.6'`. 

I will clean up inconsistency in another PR about this later if you are okay as is.